sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/NonAzureOpenAIClientImpl.java

```diff
@@ -870,4 +861,32 @@ public Response<BinaryData> generateImageWithResponse(
             Context.NONE
         );
     }
+
+    /**
+     * This method injects the modelId in the request body for requests against nonAzure OpenAI. Unlike Azure OpenAI,
+     * the service expects this value in the body of the request, whereas Azure OpenAI passes it as part of the
+     * path of the request.
+     *
+     * @param inputJson JSON submitted by the client
+     * @param modelId The LLM model ID to be injected in the JSON
+     * @return
+     */
+    private static BinaryData addModelIdJson(BinaryData inputJson, String modelId) {
+        JsonNode jsonNode = null;
+        try {
+            jsonNode = jsonMapper.readTree(inputJson.toString());
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
```

Comment:
The sync methods have to throw and async methods have to return `Mono.error()`. So, it might be best to catch the exception in the methods above and this method should just throw the exception.

Reply:
Gotcha, so what I did from what I understood:
- async: `try` return as usual / `catch` return `Mono.error(e)`
- sync: `try` return as usual / `catch` re throw `new RunTimeException(e)`

Where, `e` is a  `JsonProcessingException`